src/redux/reducers/dex.js

```diff
@@ -0,0 +1,59 @@
+import { handleActions, combineActions } from 'redux-actions';
+import { dexActions } from '../actions';
+
+const initialState = {
+  loading: false,
+  pools: null,
+  reserves: null,
+};
+
+const dexReducer = handleActions(
+  {
+    [combineActions(
+      dexActions.getPools.call,
+      dexActions.addLiquidity.call,
+      dexActions.swapTokensForExactTokens.call,
+      dexActions.swapExactTokensForTokens.call,
+      dexActions.getDexReserves.call,
+    )]: (state) => ({
+      ...state,
+      loading: true,
+    }),
+
+    [combineActions(
+      dexActions.swapExactTokensForTokens.success,
+      dexActions.swapExactTokensForTokens.failure,
+      dexActions.swapTokensForExactTokens.success,
+      dexActions.swapTokensForExactTokens.failure,
+      dexActions.getDexReserves.success,
+      dexActions.getDexReserves.failure,
+      dexActions.getDexReserves.call,
+      dexActions.getPools.success,
+      dexActions.getPools.failure,
+      dexActions.addLiquidity.success,
+      dexActions.addLiquidity.failure,
+    )]: (state) => ({
+      ...state,
+      loading: initialState.loading,
+    }),
+
+    [dexActions.getPools.call]: (state) => ({
+      ...state,
+      pools: null,
+    }),
+
+    [dexActions.getPools.success]: (state, action) => ({
+      ...state,
+      pools: action.payload,
+    }),
+
+    [dexActions.getDexReserves.success]: (state, action) => ({
+      ...state,
+      reserves: { [action.payload.asset1Number + action.payload.asset2Number]: action.payload, ...state.reserves },
```

Comment:
Won't this result in conflicts? What if we have pool for asset numbers `1` and `4` and a pool for assets `2` and `3`? They both will sum to key `5`.

Reply:
asset1Number should be string, I want to make strings 14 and 23, but i will check if there is always string

Reply:
That doesnt change it. If asset1 is `1` and asset2 is `21` it will conflict with assets `12` and `1`